src/imars3d/backend/corrections/denoise.py

```diff
@@ -215,6 +220,8 @@ class denoise(param.ParameterizedFunction):
         The sigma of the spatial space, only valid for 'bilateral' method.
     max_workers: int = 0
         The number of cores to use for parallel processing, default is 0, which means using all available cores.
+    tqdm_class:
```

Comment:
sphinx does not perform type infer for class members, so please specify the data type as the other param based class members.

Reply:
What type hint should I use for sphinx on tqdm_class? `panel.widgets.Tqdm` does [PIMPL](https://www.codeproject.com/articles/17536/use-of-pimpl-design-pattern) with `tqdm.auto.tqdm` which I then wrap with an extension `imars3d.ui.widgets.Tqdm`. I'm not sure what the right answer is here for the type hint? I could define a custom type just for annotation, but I don't have a better solution.

Reply:
How about `Union[panel.widgets.Tqdm, imars3d.ui.widgets.Tqdm]`?  This is mostly for sphinx to show the input type, and I don't think any interpreter will actually use these information during runtime.